homeassistant/components/media_player/__init__.py

```diff
@@ -304,109 +309,76 @@ def clear_playlist(hass, entity_id=None):
     hass.services.call(DOMAIN, SERVICE_CLEAR_PLAYLIST, data)
 
 
-def setup(hass, config):
+@asyncio.coroutine
+def async_setup(hass, config):
     """Track states and offer events for media_players."""
     component = EntityComponent(
         logging.getLogger(__name__), DOMAIN, hass, SCAN_INTERVAL)
 
     hass.http.register_view(MediaPlayerImageView(component.entities))
 
-    component.setup(config)
+    yield from component.async_setup(config)
 
-    descriptions = load_yaml_config_file(
-        os.path.join(os.path.dirname(__file__), 'services.yaml'))
+    descriptions = yield from hass.loop.run_in_executor(
+        None, load_yaml_config_file, os.path.join(
+            os.path.dirname(__file__), 'services.yaml'))
 
-    def media_player_service_handler(service):
+    @asyncio.coroutine
+    def async_service_handler(service):
         """Map services to methods on MediaPlayerDevice."""
         method = SERVICE_TO_METHOD[service.service]
-
-        for player in component.extract_from_service(service):
-            getattr(player, method)()
-
-            if player.should_poll:
-                player.update_ha_state(True)
+        target_players = component.async_extract_from_service(service)
+
+        update_tasks = []
+        for player in target_players:
+            kwargs = {}
+            if service.service == SERVICE_VOLUME_SET:
+                kwargs['volume'] = service.data.get(ATTR_MEDIA_VOLUME_LEVEL)
```

Comment:
Good point. I think we should change the parameters to match the service attributes. This should not affect the user. This will have to be done for all platforms though. Having the same names for attributes and parameters will make it more clear and save us confusion and bugs.